Implement stub for nucleotide-count

kytrinyx opened the pull request:
I went with the simplest possible interpretation of how to translate the data from problem-specifications into a chapel test.

Let me know if the template should be tweaked in any way in order to be more idiomatic chapel and I can regenerate the test file.

Here are some reference implementations from other languages:
- https://github.com/exercism/elixir/blob/main/exercises/practice/nucleotide-count/.meta/example.ex
- https://github.com/exercism/go/blob/main/exercises/practice/nucleotide-count/.meta/example.go
- https://github.com/exercism/ruby/blob/main/exercises/practice/nucleotide-count/.meta/example.rb

Closes #15

kytrinyx committed:
Implement stub for nucleotide-count

lucaferranti commented on exercises/practice/nucleotide-count/test/tests.chpl:
```diff
@@ -0,0 +1,30 @@
+}
+
+proc testStrandWithInvalidNucleotides(test : borrowed Test) throws {
+  try {
```
I think this is the older version of the template? A slightly better version

```chapel
proc testCaseUnderTest(test : borrowed Test) throws {
  try {
    steps(0); // Placeholder for function call here
    throw new AssertionError("Function was expected to throw an error");
  } catch e : IllegalArgumentError {
    test.assertEqual(e.message(), "Expected message here"); // placeholder for expected error message here
  } catch e {
    throw e;
  }
}
```

basically this rethrows the exception if it's not of the right kind.

kytrinyx replied:
I must have copy/pasted the wrong one. Updated!

kytrinyx committed:
Use improved template for nucleotide-count error case

lucaferranti committed:
add reference solution

lucaferranti commented:
the `AssertionError` exception is defined in the submodule `TestError` and not re-exported by default, hence when testing for exceptions we should add the line `use UnitTest.TestError;` (sorry this was easy to miss, I should have highlighted it).

We could also just always have that line, might make the generator slightly easier.

kytrinyx commented:
I did indeed complexly miss that :)

I can tweak the generators no problem.

kytrinyx committed:
Extract variable in nucleotide-count

The test suite had an unbearably long line, so I extracted the
strand into a variable for readability and regenerated the
test suite.

kytrinyx commented:
This looks great! I extracted the strand into a variable in the test suite for readability.

If you're happy with this I'd say approve and merge!

lucaferranti commented:
I am happy with this, but if you want to extract the strand in a variable also in the last test (the one that tests the error), that would be fine too.

kytrinyx commented:
Yepp, let's do that.

kytrinyx committed:
Extract variable in nucleotide-count

This extracts the strand variable in the final test of the
nucleotide-count exercise, for consistency with the rest of
the test suite.

kytrinyx commented:
Done.
